src/components/modal/modal.js

```diff
@@ -155,10 +156,14 @@ class Modal extends React.Component {
       // it was added as the Portal library we use messes up the callstack
       // when assigning the ref and triggering componentDidUpdate
       setTimeout(() => {
+        clearTimeout(this.openTimeout);
+        this.openTimeout = setTimeout(() => { this.setState({ state: 'open' }); }, 500);
         this.onOpening; // eslint-disable-line no-unused-expressions
       });
       _window.addEventListener('keyup', this.closeModal);
     } else if (!this.props.open && this.listening) {
+      clearTimeout(this.openTimeout);
+      this.openTimeout = setTimeout(() => { this.setState({ state: 'closed' }); }, 500);
```

Comment:
could `500` be pulled out into a constant in the file? and shared between this and the animations?